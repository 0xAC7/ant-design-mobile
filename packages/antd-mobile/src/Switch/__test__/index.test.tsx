import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Switch } from '..'

describe('Switch', () => {
  it('snapshot', () => {
    const component = shallow(<Switch checked disabled data-y="111" />)
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('normal', () => {
    const component = shallow(<Switch checked disabled />)

    expect(component.find('input').prop('checked')).toBeTruthy()
    expect(component.find('input').prop('disabled')).toBeTruthy()
    expect(component.find('input').prop('value')).toBe('on')

    const component2 = shallow(<Switch checked={false} />)
    expect(component2.find('input').prop('checked')).toBeFalsy()
    expect(component2.find('input').prop('disabled')).toBeFalsy()
    expect(component2.find('input').prop('value')).toBe('off')
  })
})
