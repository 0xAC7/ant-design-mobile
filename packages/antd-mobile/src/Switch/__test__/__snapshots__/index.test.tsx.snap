// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot 1`] = `
<label
  className="amd-switch"
  data-y="111"
>
  <input
    checked={true}
    className="amd-switch-checkbox"
    disabled={true}
    name=""
    onChange={[Function]}
    type="checkbox"
    value="on"
  />
  <div
    className="checkbox checkbox-disabled"
  />
</label>
`;
